src/lib/marshal/workflow/types.ts

```diff
@@ -43,10 +43,16 @@ export type ChannelStepData<A extends MaybeWithAnnotation = unknown> =
 
 /* Batch step */
 
+export enum BatchWindowType {
+  Sliding,
+  Fixed,
+}
+
 type BatchStepSettings = {
   batch_key?: string;
   batch_until_field_path?: string;
   batch_window?: Duration;
+  batch_window_type?: BatchWindowType;
```

Comment:
```suggestion
  batch_window_type: "fixed" | "sliding";
```
A couple thoughts:
* Continuing from my comment above, I think we want to type this as string enum values, or just string literals(?) like we've done for `batch_order`. 
* From the control side PR I saw, this field is always present with a value because we always fall back to a default?